mixins/grid_layout.go

```diff
@@ -0,0 +1,123 @@
+package mixins
+
+import (
+  "github.com/google/gxui"
+  "github.com/google/gxui/math"
+  "github.com/google/gxui/mixins/base"
+)
+
+type Cell struct {
+  x, y, w, h int
+}
+
+func (c Cell) AtColumn(x int) bool {
+  return c.x <= x && c.x+c.w >= x
+}
+
+func (c Cell) AtRow(y int) bool {
+  return c.y <= y && c.y+c.h >= y
+}
+
+type GridLayoutOuter interface {
+  base.ContainerOuter
+}
+
+type GridLayout struct {
+  base.Container
+
+  outer GridLayoutOuter
+
+  grid map[gxui.Control]Cell
+  rows int
+  columns int
+}
+
+func (l *GridLayout) Init(outer GridLayoutOuter, theme gxui.Theme) {
+  l.Container.Init(outer, theme)
+  l.outer = outer
+  l.grid = make(map[gxui.Control]Cell)
+
+	// Interface compliance test
+  _ = gxui.GridLayout(l)
+}
+
+func (l *GridLayout) LayoutChildren() {
+  s := l.outer.Size().Contract(l.outer.Padding())
+	o := l.outer.Padding().LT()
+
+  cw, ch := s.W / l.columns, s.H / l.rows
```

Comment:
As this math is integer, we are likely to have gaps appear on the right / bottom side of the container when we drop the remainder of these divides. Perhaps it would be better to keep the arithmetic as `float32` or `float64` and cast back to int before laying out the children?